src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -2166,4 +2306,651 @@ void CorProfiler::GetAssemblyAndSymbolsBytes(BYTE** pAssemblyArray, int* assembl
+      auto functionInfo = new FunctionInfo(caller);
+      hr = functionInfo->method_signature.TryParse();
+      if (FAILED(hr)) {
+        Warn("The method signature: ", functionInfo->method_signature.str(), " cannot be parsed.");
+        delete functionInfo;
+        enumIterator = ++enumIterator;
+        continue;
+      }
+
+      // Compare if the current mdMethodDef contains the same number of arguments as the instrumentation target
+      const auto numOfArgs = functionInfo->method_signature.NumberOfArguments();
+      if (numOfArgs != integration.replacement.target_method.signature_types.size() - 1) {
+        Debug("The caller for the methoddef: ", integration.replacement.target_method.method_name, " doesn't have the right number of arguments.");
+        delete functionInfo;
+        enumIterator = ++enumIterator;
+        continue;
+      }
+
+      // Compare each mdMethodDef argument type to the instrumentation target
+      bool argumentsMismatch = false;
+      const auto methodArguments = functionInfo->method_signature.GetMethodArguments();
+      Debug("Comparing signature for method: ", integration.replacement.target_method.type_name, ".", integration.replacement.target_method.method_name);
+      for (unsigned int i = 0; i < numOfArgs; i++) {
+        const auto argumentTypeName = methodArguments[i].GetTypeTokName(metadata_import);
+        const auto integrationArgumentTypeName = integration.replacement.target_method.signature_types[i + 1];
+        Debug("  -> ", argumentTypeName, " = ", integrationArgumentTypeName);
+        if (argumentTypeName != integrationArgumentTypeName && integrationArgumentTypeName != "_"_W) {
+          argumentsMismatch = true;
+          break;
+        }
+      }
+      if (argumentsMismatch) {
+        Debug("The caller for the methoddef: ", integration.replacement.target_method.method_name, " doesn't have the right type of arguments.");
+        delete functionInfo;
+        enumIterator = ++enumIterator;
+        continue;
+      }
+
+      // As we are in the right method, we gather all information we need and stored it in to the ReJIT handler.
+      auto moduleHandler = rejit_handler->GetOrAddModule(module_id);
+      moduleHandler->SetModuleMetadata(module_metadata);
+      auto methodHandler = moduleHandler->GetOrAddMethod(methodDef);
+      methodHandler->SetFunctionInfo(functionInfo);
+      methodHandler->SetMethodReplacement(new MethodReplacement(integration.replacement));
+
+      // Store module_id and methodDef to request the ReJIT after analyzing all integrations.
+      vtModules.push_back(module_id);
+      vtMethodDefs.push_back(methodDef);
+      
+      bool caller_assembly_is_domain_neutral = runtime_information_.is_desktop() && corlib_module_loaded && module_metadata->app_domain_id == corlib_app_domain_id;
+
+      Info("Enqueue for ReJIT [ModuleId=", module_id,
+           ", MethodDef=", HexStr(&methodDef, sizeof(mdMethodDef)), 
```

Comment:
For a methodDef, the `HexStr` is emitting a string like "f2060006" instead of the readable "060006F2" (notice the leading 06 to indicate the MethodDef token) when documenting the "Enqueue for ReJIT" information. I'm not sure if this is a general enough case, but at the very least can this `HexStr` result be reversed for readability of the token?

Reply:
Yeah for sure, I'm used to read it like `HexStr` but whatever reverse or not reverse is fine for me.

Reply:
done in e09e05f